pkg/registry/apis/folders/register.go

```diff
@@ -302,6 +303,14 @@ func (b *FolderAPIBuilder) validateOnCreate(ctx context.Context, id string, obj
 		}
 	}
 
+	if !util.IsValidShortUID(id) {
+		return dashboards.ErrDashboardInvalidUid
+	}
+
+	if util.IsShortUIDTooLong(id) {
+		return dashboards.ErrDashboardUidTooLong
+	}
+
```

Comment:
Folder API server is not checking for `IsValidShortUID` neither `IsShortUIDTooLong`